nilearn/_utils/niimg_conversions.py

```diff
@@ -418,124 +420,24 @@ def concat_niimgs(
-
-    target_fov = "first" if auto_resample else None
-
-    # We remove one to the dimensionality because of the list is one dimension.
-    ndim = None
-    if ensure_ndim is not None:
-        ndim = ensure_ndim - 1
-
-    # If niimgs is a string, use glob to expand it to the matching filenames.
-    niimgs = resolve_globbing(niimgs)
-
-    # First niimg is extracted to get information and for new_img_like
-    first_niimg = None
-
-    iterator, literator = itertools.tee(iter(niimgs))
-    try:
-        first_niimg = check_niimg(next(literator), ensure_ndim=ndim)
-    except StopIteration:
-        raise TypeError("Cannot concatenate empty objects")
-    except DimensionError as exc:
-        # Keep track of the additional dimension in the error
-        exc.increment_stack_counter()
-        raise
-
-    # If no particular dimensionality is asked, we force consistency wrt the
-    # first image
-    if ndim is None:
-        ndim = len(first_niimg.shape)
-
-    if ndim not in [3, 4]:
-        raise TypeError(
-            "Concatenated images must be 3D or 4D. You gave a "
-            f"list of {ndim}D images"
-        )
-
-    lengths = [first_niimg.shape[-1] if ndim == 4 else 1]
-    for niimg in literator:
-        # We check the dimensionality of the niimg
-        try:
-            niimg = check_niimg(niimg, ensure_ndim=ndim)
-        except DimensionError as exc:
-            # Keep track of the additional dimension in the error
-            exc.increment_stack_counter()
-            raise
-        lengths.append(niimg.shape[-1] if ndim == 4 else 1)
-
-    target_shape = first_niimg.shape[:3]
-    if dtype is None:
-        dtype = _get_data(first_niimg).dtype
-    data = np.ndarray(target_shape + (sum(lengths),), order="F", dtype=dtype)
-    cur_4d_index = 0
-    for index, (size, niimg) in enumerate(
-        zip(
-            lengths,
-            iter_check_niimg(
-                iterator,
-                atleast_4d=True,
-                target_fov=target_fov,
-                memory=memory,
-                memory_level=memory_level,
-            ),
-        )
-    ):
-        if verbose > 0:
-            if isinstance(niimg, str):
-                nii_str = f"image {niimg}"
-            else:
-                nii_str = f"image #{index}"
-            print(f"Concatenating {index + 1}: {nii_str}")
-
-        data[..., cur_4d_index : cur_4d_index + size] = _get_data(niimg)
-        cur_4d_index += size
+    """Concatenate a list of 3D/4D niimgs of varying lengths."""
+    warnings.warn(
+        (
+            "Import 'concat_niimgs' from 'nilearn._utils.niimg_conversions'\n"
+            "will be deprecated in version 0.13.\n"
+            "Importing from 'nilearn._utils.niimg_conversions'\n"
+            "will be possible at least until release 0.13.0.\n"
+            "Please import this function from 'nilearn.image' instead."
+        ),
+        DeprecationWarning,
+    )
```

Comment:
Is this deprecation needed? `concat_niimgs` is a private function and not exposed in the docs, it is rather aliased as `concat_imgs` in nilearn.image so public API is not actually changing so it shouldn't affect "proper" user code

Reply:
agreed this is me being extra cautious

am fine removing the warning if you think this is an overkill: it is true that we have a multiplication of new deprecation warnings we'll have to take care of in a couple of versions and it may be better to stick to the ones that are strictly needed.

Reply:
Ok with me to get rid of that one.

Reply:
Wait: just to be clear

@ymzayek you are arguing to remove `concat_niimgs` entirely from `nilearn/_utils/niimg_conversions.py`, not just the warning, right?

- Same for `copy_img` in `nilearn/_utils/niimg.py` in PR #4166?

Because otherwise we are left with private functions that are just "redirects" to public ones which feels a bit futile.

Reply:
Yes I would remove it